types/time_test.go

```diff
@@ -1727,6 +1786,94 @@ func TestParseTimeFromInt64(t *testing.T) {
 	require.Equal(t, 00, output.Microsecond())
 }
 
+func TestParseTimeFromFloat64(t *testing.T) {
+	sc := mock.NewContext().GetSessionVars().StmtCtx
+	sc.IgnoreZeroInDate = true
+
+	cases := []struct {
+		f   float64
+		t   byte // type: date or datetime.
+		Y   int
+		M   int
+		D   int
+		h   int
+		m   int
+		s   int
+		ms  int
```

Comment:
`ms` means normally milliseconds (1/1000 second) while microseconds (1/1000000) is normally `us`.
```suggestion
		us  int
```

Reply:
Right. I kept reminding myself the difference between ms and us all the time. And yet I made this mistake.

There are other mis-uses of "ms", I will fix them as well.